From eea784acdaeab99c890014cdbf97c68f8670a4ca Mon Sep 17 00:00:00 2001
From: "Rywacki, Kamil" <kamil.rywacki@intel.com>
Date: Thu, 16 Jul 2020 10:31:58 +0200
Subject: [PATCH] PONRTSYS-7773:
 Co-existence_of_Strict_Priority_and_WRR_scheduling; LAN port drops

- Always update DS queues mapping
---
 drivers/net/datapath/pon_qos/pon_qos_tc_qos.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c b/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
index 3d0e45f0fd0f..534960c1f4a5 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_qos.c
@@ -928,6 +928,12 @@ int pon_qos_update_qmap(struct net_device *dev,
 
 		ret = pon_qos_config_cpu_port(qid, sch->port, tc, en);
 	} else {
+		ret = pon_qos_map_qid(qid, sch->port, tc, en);
+		if (ret != 0) {
+			netdev_err(dev, "%s: qid map err\n", __func__);
+			return -EFAULT;
+		}
+
 		if (sch->ds && q_tc->indev) {
 			/* Get the queue id from the handle. In PON pq = 1 is
 			 * the one with the highest priority e.g.
@@ -938,12 +944,6 @@ int pon_qos_update_qmap(struct net_device *dev,
 			return pon_qos_force_tc(q_tc->indev, pqn, en);
 		}
 
-		ret = pon_qos_map_qid(qid, sch->port, tc, en);
-		if (ret != 0) {
-			netdev_err(dev, "%s: qid map err\n", __func__);
-			return -EFAULT;
-		}
-
 		if (!q_tc->indev && sch->ds && sch->dev == dev) {
 			ret = pon_qos_reassign_tc(sch, tc, en);
 			if (ret != 0)
